src/rust/ensogl/src/display/shape/compound/port.rs

```diff
@@ -0,0 +1,144 @@
+//! This module defines the shapes required for drawing node ports.
+
+use crate::data::color::*;
+use crate::display::shape::*;
+use crate::display::shape::primitive::def::modifier::immutable::*;
+use crate::display::shape::primitive::def::primitive::*;
+use crate::prelude::*;
+
+use crate::display::shape::primitive::def::class::ShapeOps;
+use crate::math::circle_segment::CircleSegment;
+use crate::math::topology::unit::AngleOps;
+use crate::math::topology::unit::Distance;
+use crate::math::topology::unit::PixelDistance;
+use crate::math::topology::unit::Pixels;
+use crate::math::topology::unit::{Angle, Degrees};
+
+use nalgebra::Vector2;
+
+
+
+// ===========================
+// === Port Specification ===
+// ===========================
+
+/// Indicates whether a port is incoming or outgoing.
+#[derive(Clone,Copy,Debug,Eq,PartialEq)]
+pub enum PortDirection {
```

Comment:
We try to name things for qualified usage. So the API will be consumed as `use something::port` and then `port::PortDirection`, which does not make much sense. Thus such things should be renamed to `Direction`. I know that sometimes in our codebase the names are wrong, we are slowly fixing that.